java-client/src/main/java/org/opensearch/client/opensearch/OpenSearchHttpClient.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.client.opensearch;
+
+import java.io.IOException;
+import java.net.http.HttpResponse;
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import org.opensearch.client.ApiClient;
+import org.opensearch.client.opensearch._types.ErrorResponse;
+import org.opensearch.client.opensearch._types.OpenSearchException;
+import org.opensearch.client.opensearch.request.GetHttpRequest;
+import org.opensearch.client.transport.Endpoint;
+// import org.opensearch.client.opensearch.request.GetHttpResponse;
+import org.opensearch.client.transport.JsonEndpoint;
+import org.opensearch.client.transport.OpenSearchTransport;
+import org.opensearch.client.transport.TransportOptions;
+import org.opensearch.client.transport.endpoints.EndpointWithResponseMapperAttr;
+import org.opensearch.client.util.ObjectBuilder;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class OpenSearchHttpClient extends ApiClient<OpenSearchTransport, OpenSearchHttpClient>{
+
+    public OpenSearchHttpClient(OpenSearchTransport transport) {
+        super(transport, null);
+    }
+
+    public OpenSearchHttpClient(OpenSearchTransport transport, @Nullable TransportOptions transportOptions) {
+        super(transport, transportOptions);
+    }
+
+    // ----- Endpoint: get
+
+    /**
+     * Makes a GET Request.
+     * 
+     *
+     */
+
+    public String get(GetHttpRequest request)
```

Comment:
We should return the valid response object, as per request type, I think:

```suggestion
    public GetHttpResponse  get(GetHttpRequest request)
```

Reply:
I agree we probably should return `GetHttpResponse` and the likes (in an object structure) instead of a string.

Reply:
And we can add toString() for these responses.

Reply:
What is we used the `Response` class itself from the RestClient? I meant to return the original response directly without parsing the content through a deserializer. Do we still need additional classes to handle the Response? @reta

Reply:
I think we do, the response contains: status code, body (optionally) and headers (optionally). Usually, as you mentioned, the deserializer is needed, so the response class could provide the methods to get the payload in a way client thinks it should be, including (but not limited to) `String`.